pymeasure/instruments/siglenttechnologies/siglent_spdbase.py

```diff
@@ -138,137 +232,23 @@ def enable_local_interface(self, enable: bool = True):
-        """Measure the channel output voltage.
-
-        :type: float
-        """
-    )
-
-    current = Instrument.measurement(
-        "MEAS:CURR? CH{channel}",
-        """Measure the channel output current.
-
-        :type: float
-        """
-    )
-
-    power = Instrument.measurement(
-        "MEAS:POWE? CH{channel}",
-        """Measure the channel output power.
-
-        :type: float
-        """
-    )
-
-    current_limit = Instrument.control(
-        "CH{channel}:CURR?", "CH{channel}:CURR %g",
-        """Control the output current configuration of the channel.
-
-        :type : float
-        """,
-        validator=truncated_range,
-        values=[0, 8],
-        dynamic=True
-    )
-
-    voltage_setpoint = Instrument.control(
-        "CH{channel}:VOLT?", "CH{channel}:VOLT %g",
-        """Control the output voltage configuration of the channel.
-
-        :type : float
-        """,
-        validator=truncated_range,
-        values=[0, 16],
-        dynamic=True
-    )
-
-    def ask(self, cmd):
-        return self.instrument.ask(cmd.format(channel=self.channel))
-
-    def write(self, cmd):
-        self.instrument.write(cmd.format(channel=self.channel))
-
-    def values(self, cmd, **kwargs):
-        """ Reads a set of values from the instrument through the adapter,
-        passing on any key-word arguments.
-        """
-        return self.instrument.values(cmd.format(channel=self.channel), **kwargs)
-
-    def check_errors(self):
-        return self.instrument.check_errors()
-
-    def enable_output(self, enable: bool = True):
-        """Enable the channel output.
-
-        :type: bool
-            ``True``: enables the output
-            ``False``: disables it
-        """
-        self.instrument.selected_channel = self.channel
-        self.write(f'OUTP CH{self.channel},{("OFF","ON")[enable]}')
-
-    def enable_timer(self, enable: bool = True):
-        """Enable the channel timer.
-
-        :type: bool
-            ``True``: enables the timer
-            ``False``: disables it
-        """
-        self.write(f'TIME CH{self.channel},{("OFF","ON")[enable]}')
-
-    def configure_timer(self, step, voltage, current, duration):
-        """Configure the timer step.
-
-        :param step:
-            int: index of the step to save the configuration
-        :param voltage:
-            float: voltage setpoint of the step
-        :param current:
-            float: current limit of the step
-        :param duration:
-            int: duration of the step in seconds
-        """
-        step = strict_discrete_range(step, [1, 5], 1)
-        voltage = truncated_range(voltage, self.voltage_range)
-        current = truncated_range(current, self.current_range)
-        duration = truncated_range(duration, [0, 10000])
-        self.write(f'TIME:SET CH{self.channel},{step:d},{voltage:1.3f},{current:1.3f},{duration:d}')
+        self.write(f'MODE:SET {("4W" if enable else "2W")}')
```

Comment:
Neither do you need the parenthesis here.
```suggestion
        self.write(f'MODE:SET {"4W" if enable else "2W"}')
```

In one example it was needed to group the strings `"A" + ("B" if True else "C")` in order to combine AB or AC, but normally it is sufficient to write `print("A" if True else "B")`.